src/transaction.cpp

```diff
@@ -755,16 +754,18 @@ int wsrep::transaction::after_rollback()
     if (is_streaming() && bf_aborted_in_total_order_)
     {
         remove_fragments_in_storage_service_scope(lock);
-        streaming_context_.cleanup();
-    }
-
-    if (is_streaming() && state() != s_must_replay)
-    {
-        streaming_context_.cleanup();
     }
 
     if (state() == s_aborting)
     {
+        if (is_streaming())
```

Comment:
Comment is confusing. Ok, we are in state aborting and we are streaming but where that replay comes in here?

Reply:
When after_rollback() hook is called, transaction is either in s_aborting or s_must_replay state.

Reply:
ah so it is on assertion, thanks